site/docs/integrations/astro.md

```diff
@@ -0,0 +1,65 @@
+---
+title: Astro
+parent: integrations
+---
+
+# Astro
+
+Integrating Vanilla Extract with [Astro](https://astro.build) is done with the help of the [Vite plugin][vite integration].
+
+## Installation
+
+```bash
+npm install @vanilla-extract/css
+npm install @vanilla-extract/vite-plugin --save-dev
+```
+
+## Setup
+
+Add Vanilla Extract Vite plugin to the Astro configuration:
+
+```js
+import { defineConfig } from 'astro/config';
+import { vanillaExtractPlugin } from '@vanilla-extract/vite-plugin';
+
+// https://astro.build/config
+export default defineConfig({
+  vite: {
+    plugins: [vanillaExtractPlugin()]
+  }
+});
+```
+
+You'll then be able to use `style` and other APIs in `.css.ts` files.
+
+```ts
+// button.css.ts
+
+import { style } from '@vanilla-extract/css';
+
+export const button = style({
+  padding: '0.5rem 1rem',
+  border: 'none',
+  borderRadius: '0.25rem',
+  color: 'white',
+  background: '#333'
+});
+```
+
+And now you can reference styles in your Astro component:
+
+```tsx
+//  Button.astro
```

Comment:
Bit of a hack, but it's not stupid if it works 😅 

<img width="433" alt="Button.astro" src="https://github.com/vanilla-extract-css/vanilla-extract/assets/3297808/766a12c2-0910-4f4e-b520-b7699ee2c794">